src/routes/(app)/product/[id]/+page.svelte

```diff
@@ -129,6 +129,22 @@
 	}
 
 	let selectedVariations: Record<string, string> = {};
+	$: if (data.product.hasVariations && data.product.variationLabels) {
+		customAmount = toCurrency(
+			data.currencies.main,
+			data.product.price.amount,
+			data.product.price.currency
+		);
+		for (const [key, value] of Object.entries(selectedVariations)) {
+			customAmount += toCurrency(
+				data.currencies.main,
+				data.product.variations?.find(
+					(variation) => variation.name === key && variation.value === value
+				)?.price ?? 0,
+				data.product.price.currency
+			);
+		}
```

Comment:
use `sumCurrencies` here - it takes care of some details with floating point precision